sigauth/signature.go

```diff
@@ -38,17 +40,25 @@ import (
 type SignatureAuth struct {
 	serviceRegManager *authservice.ServiceRegManager
 
-	serviceKey *rsa.PrivateKey
+	serviceKey authutils.PrivateKey
 }
 
 // Sign generates and returns a signature for the provided message
 func (s *SignatureAuth) Sign(message []byte) (string, error) {
-	hash, err := authutils.HashSha256(message)
-	if err != nil {
-		return "", fmt.Errorf("error hashing message: %v", err)
-	}
+	var opts crypto.SignerOpts
+	var err error
 
-	signature, err := rsa.SignPSS(rand.Reader, s.serviceKey, crypto.SHA256, hash, nil)
+	switch s.serviceKey.(type) {
+	case *rsa.PrivateKey, *ecdsa.PrivateKey:
+		message, err = authutils.HashSha256(message)
+		if err != nil {
+			return "", fmt.Errorf("error hashing message: %v", err)
+		}
+		opts = &rsa.PSSOptions{Hash: crypto.SHA256}
+	case ed25519.PrivateKey:
+		opts = &rsa.PSSOptions{Hash: 0}
+	}
```

Comment:
It would be nice if we could abstract these types of type checks behind the `PrivateKey`/`PublicKey` interfaces. I assume this is currently not being done to avoid breaking changes, but if it will simplify things I think we should go ahead and make breaking changes.